pkg/config/handler_test.go

```diff
@@ -0,0 +1,114 @@
+package config
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+var testStorage = &Storage{
+	Disk: &DiskConfig{
+		RootPath: "/path/on/disk",
+	},
+	GCP: &GCPConfig{
+		ProjectID: "MY_GCP_PROJECT_ID",
+		Bucket:    "MY_GCP_BUCKET",
+	},
+	Minio: &MinioConfig{
+		Endpoint:  "127.0.0.1:9001",
+		Key:       "minio",
+		Secret:    "minio123",
+		EnableSSL: false,
+		Bucket:    "gomods",
+	},
+	Mongo: &MongoConfig{
+		URL:                   "mongodb://127.0.0.1:27017",
+		CertPath:              "",
+		InsecureConn:          false,
+		DefaultDBName:         "athens",
+		DefaultCollectionName: "modules",
+	},
+	S3: &S3Config{
+		Region: "MY_AWS_REGION",
+		Key:    "MY_AWS_ACCESS_KEY_ID",
+		Secret: "MY_AWS_SECRET_ACCESS_KEY",
+		Token:  "",
+		Bucket: "MY_S3_BUCKET_NAME",
+	},
+}
+
+var testConfig = &Config{
+	GoEnv:           "development",
+	LogLevel:        "debug",
+	GoBinary:        "go",
+	GoProxy:         "direct",
+	GoGetWorkers:    10,
+	ProtocolWorkers: 30,
+	CloudRuntime:    "none",
+	TimeoutConf: TimeoutConf{
+		Timeout: 300,
+	},
+	StorageType:      "memory",
+	GlobalEndpoint:   "http://localhost:3001",
+	Port:             ":3000",
+	EnablePprof:      false,
+	PprofPort:        ":3001",
+	BasicAuthUser:    "",
+	BasicAuthPass:    "",
+	Storage:          testStorage,
+	TraceExporterURL: "http://localhost:14268",
+	TraceExporter:    "",
+	StatsExporter:    "prometheus",
+	SingleFlightType: "memory",
+	GoBinaryEnvVars:  []string{"GOPROXY=direct"},
+	SingleFlight:     &SingleFlight{},
+	SumDBs:           []string{"https://sum.golang.org"},
+	NoSumPatterns:    []string{},
+	DownloadMode:     "sync",
+	RobotsFile:       "robots.txt",
+}
+
+func TestConfigHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/config", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	h := Handler(testConfig)
+	rr := httptest.NewRecorder()
+
+	h.ServeHTTP(rr, req)
+
+	// Check the status code is what we expect.
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+
+	toCheck := &Config{}
+	if err := json.Unmarshal(rr.Body.Bytes(), toCheck); err != nil {
+		t.Error("Failed to unmarshal the reply")
+	}
+	if !reflect.DeepEqual(toCheck, testConfig) {
+		t.Error("Returned config different from original")
+	}
```

Comment:
Let's use `cmp` to check for equality, but more importantly to output a friendly Diff so that if the test fails a user knows why it failed. 

[See this line of code for an example(https://github.com/gomods/athens/blob/23672f78c933b1d7ad4b3353ce628ccfb1b988b1/pkg/config/config_test.go#L28:L32)

Reply:
Did not know about it, awesome!